crates/proof-of-sql/sol_src/proof_primitive/sumcheck/Sumcheck.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+library Sumcheck {
+    error InvalidProofLength();
+    error RoundEvaluationMismatch();
+
+    struct Proof {
+        uint256[] coefficients;
+    }
+
+    struct Subclaim {
+        uint256[] evaluationPoint;
+        uint256 expectedEvaluation;
+    }
+
+    function verifyWithoutEvaluation(
+        Proof calldata proof,
+        bytes32 transcript,
+        uint64 degree,
+        uint64 numVariables,
+        uint256 claimedSum,
+        uint256 modulus
+    ) public pure returns (Subclaim memory result, bytes32 resultTranscript) {
+        if (proof.coefficients.length != numVariables * (degree + 1)) revert InvalidProofLength();
+        result.evaluationPoint = new uint256[](numVariables);
+        transcript = keccak256(abi.encodePacked(transcript, degree, numVariables));
+        for (uint64 roundIndex = 0; roundIndex < numVariables; ++roundIndex) {
+            uint256 startIndex = roundIndex * (degree + 1);
+            transcript = keccak256(abi.encodePacked(transcript, proof.coefficients[startIndex:startIndex + degree + 1]));
+            uint256 roundEvaluationPoint = uint256(transcript);
+            result.evaluationPoint[roundIndex] = roundEvaluationPoint % modulus;
```

Comment:
how do we insure the randomness of this chosen point?

Reply:
See my response to Ryan. The point is random because it is a hash of the previous transcript along with the previous prover message.